core/src/read.jl

```diff
@@ -635,111 +635,143 @@ function PidControl(db::DB, config::Config, chunk_sizes::Vector{Int})::PidContro
+            @error "The demand(t) relationship for UserDemand $node_id of priority $p from the time table has repeated timestamps, this can not be interpolated."
+            errors = true
+        end
+    end
+    return errors
+end
 
-    # Whether the demand of a UserDemand node is given by a timeseries
-    demand_from_timeseries = BitVector()
+function UserDemand(db::DB, config::Config)::UserDemand
+    static = load_structvector(db, config, UserDemandStaticV1)
+    time = load_structvector(db, config, UserDemandTimeV1)
 
-    for node_id in node_ids
-        first_row = nothing
-        demand_itp_node_id = Vector{ScalarInterpolation}()
+    _, _, node_ids, valid = static_and_time_node_ids(db, static, time, "UserDemand")
 
-        if node_id in static_node_ids
-            push!(demand_from_timeseries, false)
-            rows = searchsorted(NodeID.(NodeType.UserDemand, static.node_id), node_id)
-            static_id = view(static, rows)
-            for p in priorities
-                idx = findsorted(static_id.priority, p)
-                demand_p = !isnothing(idx) ? static_id[idx].demand : 0.0
-                demand_p_itp = LinearInterpolation([demand_p, demand_p], trivial_timespan)
-                push!(demand_itp_node_id, demand_p_itp)
-                push!(demand, demand_p)
-            end
-            push!(demand_itp, demand_itp_node_id)
-            first_row = first(static_id)
-            is_active = coalesce(first_row.active, true)
+    if !valid
+        error("Problems encountered when parsing UserDemand static and time node IDs.")
+    end
 
-        elseif node_id in time_node_ids
-            push!(demand_from_timeseries, true)
-            for p in priorities
-                push!(demand, 0.0)
-                if p in keys(time_priority_dict)
-                    demand_p_itp, is_valid = get_scalar_interpolation(
-                        config.starttime,
-                        t_end,
-                        time_priority_dict[p],
-                        node_id,
-                        :demand;
-                        default_value = 0.0,
-                    )
-                    if is_valid
-                        push!(demand_itp_node_id, demand_p_itp)
-                    else
-                        @error "The demand(t) relationship for UserDemand #$node_id of priority $p from the time table has repeated timestamps, this can not be interpolated."
-                        errors = true
-                    end
-                else
-                    demand_p_itp = LinearInterpolation([0.0, 0.0], trivial_timespan)
-                    push!(demand_itp_node_id, demand_p_itp)
-                end
-            end
-            push!(demand_itp, demand_itp_node_id)
+    # Initialize vectors for UserDemand fields
+    priorities = get_all_priorities(db, config)
+    n = length(node_ids)
+    active = BitVector(ones(Bool, n))
+    realized_bmi = zeros(n)
+    demand = zeros(n * length(priorities))
```

Comment:
I'll open a separate issue, but to be clear, I think this would be nicer:

```julia
n_priority = length(priorities)
demand = zeros((n_priority, n_user)
allocated = fill(Inf, (n_priority, n_user)
```
This obviously means more changes in the code.